go/vt/vtgate/planbuilder/builder.go

```diff
@@ -111,9 +111,11 @@ func getConfiguredPlanner(vschema plancontext.VSchema, v3planner func(string) se
 			fallback: v3planner(query),
 		}
 		return fp.plan, nil
-	default:
-		// default is v3 plan
+	case V3:
 		return v3planner(query), nil
+	default:
+		// default is gen4 plan
+		return gen4Planner(query, Gen4), nil
 	}
```

Comment:
nit: remove case Gen4 from above.

Reply:
the gen4 case is still needed. There are variations of gen4 (left2right and greedy only) that are not handled by the default case